src/libraries/System.IO.FileSystem.AccessControl/tests/FileSystemSecurityTests.cs

```diff
@@ -409,5 +416,85 @@ public void AuditRuleType_Returns_Valid_Object()
             Type type = fileSecurity.AuditRuleType;
             Assert.Equal(typeof(FileSystemAuditRule), type);
         }
+
+        [Fact]
+        public void DirNotFound_DirectorySecurity()
+        {
+            using TempDirectory tempDirectory = new TempDirectory();
+            {
+                string longDir = CreateLongDirectory(tempDirectory.Path, create: false);
+                Assert.Throws<DirectoryNotFoundException>(() =>
+                {
+                    var security = new DirectorySecurity(longDir, AccessControlSections.Owner);
+                });
+            }
+        }
+
+        [Fact]
+        public void FileNotFound_FileSecurity()
+        {
+            using TempDirectory tempDirectory = new TempDirectory();
+            {
+                string longDir = CreateLongDirectory(tempDirectory.Path, create: false);
+                string filePath = Path.Combine(longDir, "file.txt");
+                Assert.Throws<FileNotFoundException>(() =>
+                {
+                    var security = new FileSecurity(filePath, AccessControlSections.Owner);
+                });
+            }
+        }
+
+        [Fact]
+        public void MaxLengthPath_DirectorySecurity()
+        {
+            using TempDirectory tempDirectory = new TempDirectory();
+            {
+                string longDir = CreateLongDirectory(tempDirectory.Path);
+                var security = new DirectorySecurity(longDir, AccessControlSections.Owner);
+                Assert.NotNull(security);
+            }
+        }
+
+        [Fact]
+        public void MaxLengthPath_FileSecurity()
+        {
+            using (TempDirectory tempDirectory = new TempDirectory())
+            {
+                string longDir = CreateLongDirectory(tempDirectory.Path);
+                using TempFile tempFile = new TempFile(Path.Combine(longDir, "file.txt"));
+                var security = new FileSecurity(tempFile.Path, AccessControlSections.Owner);
+                Assert.NotNull(security);
+            }
+        }
+
+        private string CreateLongDirectory(string basePath, bool create = true)
+        {
+            string fullPath = Path.Combine(basePath, GetRandomName(), GetRandomName(), GetRandomName(), GetRandomName(), GetRandomName());
+
+            if (create)
+            {
+                Directory.CreateDirectory(fullPath);
+            }
+
+            return fullPath;
+        }
+
+        private string GetRandomName()
```

Comment:
Is this something better shared? Maybe in https://github.com/dotnet/runtime/blob/master/src/libraries/Common/tests/CoreFx.Private.TestUtilities/System/IO/FileCleanupTestBase.cs

Reply:
I moved it to TempDirectory, I have access to that class.